lib/plausible_web/controllers/api/external_stats_controller.ex

```diff
@@ -96,6 +96,20 @@ defmodule PlausibleWeb.Api.ExternalStatsController do
      "The `property` parameter is required. Please provide at least one property to show a breakdown by."}
   end
 
+  @max_breakdown_limit 1000
+  defp validate_or_default_limit(%{"limit" => limit}) do
+    limit = String.to_integer(limit)
+
+    if limit <= @max_breakdown_limit do
+      {:ok, limit}
+    else
+      {:error, "Limit too large. Please use a limit not higher than #{@max_breakdown_limit}."}
+    end
```

Comment:
```suggestion
  with {limit, ""} when limit > 0 and limit <= @max_breakdown_limit <- Integer.parse(limit) do
    {:ok, limit}
  else _ -> 
    {:error, "Please provide limit as a number between 1 and #{@max_breakdown_limit}."}
  end
```

This way you're not crashing the request when someone sends `limit=foo`(the cases for non-integer and zero/negative limits could maybe use some tests too?)

Reply:
Thanks, @aerosol! Changed it and added tests: https://github.com/plausible/analytics/pull/2226/commits/1e1a47092985e75b837c1d2547511e2ffe632d2d